src/fidesops/api/v1/endpoints/connection_endpoints.py

```diff
@@ -177,12 +182,21 @@ def validate_secrets(
     schema = get_connection_secrets_validator(connection_type.value, saas_config)
 
     try:
+        connection_type = connection_config.connection_type
+        saas_config = connection_config.get_saas_config()
+        if connection_type == ConnectionType.saas and saas_config is None:
+            raise SaaSConfigNotFoundException(
+                f"A SaaS config to validate the secrets is unavailable for this connection config, please add one via {SAAS_CONFIG}"
+            )
+        schema = get_connection_secrets_validator(connection_type.value, saas_config)
+        logger.info(
+            f"Validating secrets on connection config with key '{connection_config.key}'"
+        )
         connection_secrets = schema.parse_obj(request_body)
     except ValidationError as e:
-        raise HTTPException(
-            status_code=422,
-            detail=e.errors(),
-        )
+        raise HTTPException(status_code=422, detail=e.errors())
+    except SaaSConfigNotFoundException as e:
+        raise HTTPException(status_code=422, detail=e.message)
     return connection_secrets
```

Comment:
Question about the mailchimp config - it seems like secrets validation returns "succeeded" regardless of username and api_key.  Is validation happening properly there, or is this something specific related to the test account?

Reply:
I wasn't handing non HTTP status codes correctly, the connection test fails now when providing incorrect credentials.